app/forms/vehicle_form/purchase_date_form.rb

```diff
@@ -4,75 +4,59 @@ class PurchaseDateForm
 
     form_for Vehicle
 
-    DATE_PARTS = %i[purchased_on_year purchased_on_month purchased_on_day].freeze
-
-    attr_accessor(*DATE_PARTS)
-    attr_accessor :mode
+    attr_accessor :purchased_on_year, :purchased_on_month, :purchased_on_day
+    attr_accessor :journey
     attr_writer :purchased_on
 
     validate :date_is_in_the_future
 
-    validates :purchased_on,
-              date: { allow_nil: true },
-              presence: { unless: :draft_and_not_incomplete_date? }
-
-    def purchased_on
-      return @purchased_on if @purchased_on.present?
-      return incomplete_date if date_values.any?(&:blank?)
-      return unless Date.valid_date?(*date_values.map(&:to_i))
-
-      @purchased_on = attributes[:purchased_on] = Date.new(*date_values.map(&:to_i))
-    end
+    validates :purchased_on, presence: true, unless: :draft_and_not_partially_complete_date?
+    validates :purchased_on, date: { not_in_the_future: true }, allow_nil: true
```

Comment:
If you're using `validates :purchased_on, date: { not_in_the_future: true } ...` (which I think you should), then `validate :date_is_in_the_future` becomes redundant. So:

Remove `validate :date_is_in_the_future` and the method `date_is_in_the_future`

Reply:
Also `error_message_for` can be removed as it's only used by `date_is_in_the_future`